aws/rust-runtime/aws-config/src/connector.rs

```diff
@@ -44,7 +44,37 @@ pub fn default_connector(
 }
 
 /// Given `ConnectorSettings` and an `AsyncSleep`, create a `DynConnector` from defaults depending on what cargo features are activated.
-#[cfg(not(feature = "rustls"))]
+#[cfg(all(not(feature = "rustls"), feature = "native-tls"))]
+pub fn default_connector(
+    settings: &ConnectorSettings,
+    sleep: Option<Arc<dyn AsyncSleep>>,
+) -> Option<DynConnector> {
+    let hyper = base(settings, sleep).build(aws_smithy_client::conns::native_tls());
+    Some(DynConnector::new(hyper))
+}
+
+/// Given `ConnectorSettings` and an `AsyncSleep`, create a `DynConnector` from defaults depending on what cargo features are activated.
+#[cfg(all(
+    feature = "wasm",
+    target_family = "wasm",
+    target_os = "wasi",
+    not(any(feature = "rustls", feature = "native-tls"))
+))]
+pub fn default_connector(
+    _settings: &ConnectorSettings,
+    _sleep: Option<Arc<dyn AsyncSleep>>,
+) -> Option<DynConnector> {
+    Some(DynConnector::new(
+        aws_smithy_wasm::wasi_adapter::Adapter::default(),
+    ))
+}
```

Comment:
let's remove this—I think we've been bitten a lot by opting code in and out by features like this. 

I think it's OK if initially, you need to configure the connector directly to get everything to work on WASM

Reply:
moved all this logic to the wasm crate